plutus-tx/testlib/PlutusTx/Test.hs

```diff
@@ -198,6 +201,18 @@ instance (PLC.Closed uni, uni `PLC.Everywhere` Flat, Flat fun) =>
         v' <- catchAll $ getPlcNoAnn v
         toUPlc v'
 
+instance
+         ( PLC.PrettyParens (PLC.SomeTypeIn uni)
+         , PLC.GEq uni, PLC.Typecheckable uni fun
+         , PLC.Closed uni, uni `PLC.Everywhere` PrettyConst, Pretty fun
+         , uni `PLC.Everywhere` Flat, Flat fun, Default (PLC.CostingPart uni fun)
+         ) => ToTPlc (CompiledCodeIn uni fun a) uni fun where
+    toTPlc v = do
+        mayV' <- catchAll $ getPir v
+        case mayV' of
+            Nothing -> fail "No PIR available"
+            Just v' -> toTPlc v'
```

Comment:
It took me a ridiculous amount of time to figure out where to put this instance and how to implement it. The testing machinery seems scattered across multiple files with a lot of ad hoc logic etc. Not sure if we can do much about it, though.

Reply:
It's terrible and I hate it. Also not many ideas on how to improve though.